File: pinyougou-cart-interface/src/main/java/com/pinyougou/cart/service/CartService.java
package com.pinyougou.cart.service;

import com.pinyougou.pojogroup.Cart;

import java.util.List;

/**
 * @Author OliverYu
 * @Date 2018/12/17 15:25
 * @Description TODO
 */
public interface CartService {

    /**
     * 添加商品到购物车
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    public List<Cart> addGoodsToCartlist(List<Cart> cartList,Long itemId,Integer num);

    /**
     * 从redis中查询购物车
     * @param username
     * @return
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 将购物车保存到redis
     * @param username
     * @param cartList
     */
    public void saveCartToRedis(String username,List<Cart> cartList);

    /**
     * 合并购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);

}

File: pinyougou-cart-web/src/main/webapp/js/service/cartService.js
//购物车服务层
app.service('cartService',function($http){
    //购物车列表
    this.findCartList=function(){
        return $http.get('cart/findCartList.do');
    };

    //添加商品到购物车
    this.addGoodsToCartList=function(itemId,num){
        return $http.get('cart/addGoodsToCartList.do?itemId='+itemId+'&num='+num);
    };
    
    this.sum = function (cartList) {
        var totalValue = {totalNum:0,totalFee:0};
        for (var i=0;i<cartList.length;i++) {
            for (var j=0; j<cartList[i].orderItemList.length;j++) {
                totalValue.totalNum += cartList[i].orderItemList[j].num;
                totalValue.totalFee += cartList[i].orderItemList[j].totalFee;
            }
        }
        return totalValue;
    }
});